src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,139 @@
+
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE);
+    }
+
+    @Test
+    void register_allValidData_Ok() {
+        User expected = storageDao.add(user);
+        User actual = storageDao.get(user.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_allValidDataEdgeCase_Ok() {
+        user.setLogin(VALID_LOGIN_EDGE_CASE);
+        user.setPassword(VALID_PASSWORD_EDGE_CASE);
+        user.setAge(VALID_AGE_EDGE_CASE);
+        User expected = storageDao.add(user);
+        User actual = storageDao.get(user.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_sameLogin_notOk() {
+        User user2 = new User();
+        user2.setLogin(VALID_LOGIN);
+        user2.setPassword(VALID_PASSWORD_EDGE_CASE);
+        user2.setAge(VALID_AGE);
+        storageDao.add(user);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user2));
+    }
+
+    @Test
+    void register_nullLogin_notOK() {
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_login3Length_notOk() {
+        user.setLogin("abc");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_login5Length_notOk() {
+        user.setLogin("abc12");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        user.setPassword(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_password3Length_notOk() {
+        user.setPassword("123");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_password5Length_notOk() {
+        user.setPassword("12345");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ageIsNull_notOk() {
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ageIs7_notOk() {
+        user.setAge(7);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ageIs12_notOk() {
+        user.setAge(12);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ageIs17_notOk() {

```

Comment:
Probably we can call method like
`register_lowerThanRequiredAge_notOk`
It just looks more precise, what do you think?